src/tracks/playabletrack/wavetrack/ui/WaveClipTrimHandle.cpp

```diff
@@ -64,48 +71,105 @@ class WaveClipTrimHandle::AdjustBorder final : public WaveClipTrimHandle::ClipTr
       }
    }
 
+   //Search for a good snap points among all tracks, including
+   //the one to which adjusted clip belongs to, but not counting
+   //the borders of adjusted clip
+   static SnapPointArray FindSnapPoints(
+      const WaveTrack* currentTrack, 
+      WaveClip* adjustedClip,
+      const std::pair<double, double> range)
+   {
+      SnapPointArray result;
+
+      auto addSnapPoint = [&](double t, const Track* track)
+      {
+         if(t > range.second || t < range.first)
+            return;
+
+         for(const auto& snapPoint : result)
+            if(snapPoint.t == t)
+               return;
+         result.emplace_back(t, track);
+      };
+
+      if(const auto trackList = currentTrack->GetOwner())
+      {
+         for(const auto track : trackList->Any())
+         {
+            if(track == currentTrack ||
+               track->GetLinkType() == Track::LinkType::Aligned && *trackList->FindLeader(currentTrack) == track ||
+               currentTrack->GetLinkType() == Track::LinkType::Aligned && *trackList->FindLeader(track) == currentTrack)
+            {
+               //serching on the same track is a special case
+               for(const auto& clip : static_cast<const WaveTrack*>(track)->GetClips())
```

Comment:
I would prefer to iterate the intervals, as in the other branch, not the clips, for uniformity, and in case WaveTrack might add some other kind of interval in future (who knows?)

You can static_cast the Extra() pointer of each interval to WaveTrack::IntervalData* and then test for the clip you want to exclude.

Reply:
Didn't think of that. Thank you!